src/python/pants/backend/python/util_rules/lockfile_metadata.py

```diff
@@ -38,40 +53,53 @@ def from_lockfile(cls, lockfile: bytes) -> LockfileMetadata:
             elif in_metadata_block:
                 metadata_lines.append(line[2:])
 
+        error_suffix = "To resolve this error, you will need to regenerate the lockfile."
+
         if not metadata_lines:
             # TODO(#12314): Add a good error.
-            raise InvalidLockfileError("")
+            raise InvalidLockfileError(
+                "Could not find a pants metadata block in this lockfile. " + error_suffix
```

Comment:
It would be really helpful to mention the problematic lockfile in all these error messages.

> Could not find a Pants metadata block in the lockfile at <path>

Reply:
It would be. Let me figure out what it would take to be able to plumb that value in.